### PR TITLE
LPS-158632 Get the group  id from the theme display if GroupThreadLocal.getGroupId() returns 0

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
- LPS-158632 Get the group  id from the theme display if GroupThreadLocal.getGroupId() returns 0
- LPS-158632 In some cases the theme display is empty

Hi @adolfopa 

I found that when we come directly from the FriendlyURLServlet and try to obtain the urlTitle neither the GroupThreadLocal.getGroupId() or the theme display comes with the proper group id (the serviceContext.getScopeGroupId() returns the main group- liferay, not the actual site) 
So this is a posible solution without creating a serie of new methods `public String getFriendlyURL(long groupId, FileEntry fileEntry, String languageId) ` on InfoItemFriendlyURLProvider, up and down.

please feedback is welcome

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

File: modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/servlet/FriendlyURLServlet.java [hunk 1]
```diff
@@ -172,6 +174,20 @@ public Redirect getRedirect(
...
+		if (httpServletRequest.getAttribute(WebKeys.THEME_DISPLAY) == null) {
+			ThemeDisplay themeDisplay = ThemeDisplayFactory.create();
+
+			themeDisplay.setCompany(portal.getCompany(httpServletRequest));
+			themeDisplay.setPlid(portal.getControlPanelPlid(companyId));
+			themeDisplay.setRequest(httpServletRequest);
+			themeDisplay.setScopeGroupId(group.getGroupId());
+			themeDisplay.setSiteGroupId(
+				portal.getSiteGroupId(group.getGroupId()));
+
```
[the remark this answers is not in the document]
Comment (AliciaGarciaGarcia) [reply modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/servlet/FriendlyURLServlet.java]: @adolfopa done

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**